### PR TITLE
Feat(get-funding-source-interactions): Add documentation for get funding source interactions

### PR DESCRIPTION
**Pull request by chanjant**
document get Funding Source Interactions by Funding Source id api

Ticket Link: https://www.notion.so/immersve/documentation-7f7649c2b5f74f0288d8b2bdfe4fb72a?pvs=4 

Screenshot 2024-01-09 at 4 42 50 pm
Screenshot 2024-01-09 at 4 43 03 pm

### CODE REVIEW

File: openapi/endpoints/funding-sources/models/get-funding-source-interactions.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+type: object
+properties:
+  items:
+    type: array
+    items:
+      $ref: './funding-source-interaction.yaml'
+  pageInfo:
```
Comment (tonykha123) [inline openapi/endpoints/funding-sources/models/get-funding-source-interactions.yaml:7]: I wouldnt add this here yet, until the pagination PR is merged

File: openapi/endpoints/funding-sources/models/funding-source-interaction.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  context:
+    type: object
+    properties:
+      ref:
+        type: string
+        description: A context reference based on the type of the interaction. It could be a `transactionId` in the case of a payment, or an `eventId` in the case of an smart contract event.
+        example: '1000000488078'
+      type:
+        type: string
+        description: A context type for the context reference, which will depend on the interaction type.
```
Comment (tonykha123) [inline openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:53]: this doesnt sound right. any better suggestions on the description for `ref` ? @Null000 @javcode

File: openapi/endpoints/funding-sources/models/funding-source-interaction.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+type: object
+properties:
+  id:
+    type: string
+    description: id of the Funding Source Interaction.
```
Comment (tonykha123) [inline openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:5]:
```suggestion
    description: ID of the Funding Source Interaction.
```

File: openapi/endpoints/funding-sources/models/funding-source-interaction.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    example: f0220f084a182e3f4b4d605cda1d3340
+  fundingSourceId:
+    type: string
+    description: Funding Source to which the interaction belongs to.
```
Comment (tonykha123) [inline openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:9]:
```suggestion
    description: The ID of the Funding Source that this Interaction relates to.
```

File: openapi/endpoints/funding-sources/models/funding-source-interaction.yaml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    example: 2be03cd65da12fe29b8910af7b4b85e7
+  accountId:
+    type: string
+    description: The accountId to which the interaction belongs to.
```
Comment (tonykha123) [inline openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:13]:
```suggestion
    description:  The ID of the account that this Interaction relates to.
```

File: openapi/endpoints/funding-sources/models/funding-source-interaction.yaml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    description: The interaction type.
+    enum: 
+      - Deposit
+      - Withdrawal
+      - Refund
+      - Payment
+    example: Payment
+  status:
+    type: string
+    description: The current status of the interaction. It will vary depending on the interaction type.
```
Comment (tonykha123) [inline openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:26]: we should probably explain what the different common statuses mean

File: openapi/endpoints/funding-sources/models/funding-source-interaction.yaml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        type: string
+        description: A context reference based on the type of the interaction. It could be a `transactionId` in the case of a payment, or an `eventId` in the case of an smart contract event.
+        example: '1000000488078'
+      type:
+        type: string
+        description: A context type for the context reference, which will depend on the interaction type. 
+        example: transaction
+      blockNumber:
+        type: string
+        description: The chain block number in which the transactions was mined, when applicable.
```
Comment (tonykha123) [inline openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:57]: not sure about the `when applicable` part

Comment (chanjant) [reply openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:58]: should i move to an optional child object as suggested in the comment here:
https://www.notion.so/Get-Funding-Source-Interactions-By-Funding-Source-Id-1d6dc8a91a8b4da892f5bd2fa9938e42?d=264f4bf6196745cba3f622ec8350baa8&pvs=4#c8cc72bf944843e481701ba63362d80f

Comment (tonykha123) [reply openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:58]: I think you can just put (optional) imo

Comment (tonykha123) [reply openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:58]: actually when applicable should be fine

File: openapi/endpoints/funding-sources/models/funding-source-interaction.yaml (hunk 6 above)
Comment (chanjant) [reply openapi/endpoints/funding-sources/models/funding-source-interaction.yaml:27]: what are the common statuses?

**Review by ncjones — APPROVED** [review]

**Pull request merged**